opm/simulators/utils/ParallelEclipseState.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+  Copyright 2019 Equinor AS.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <config.h>
+
+#include "ParallelEclipseState.hpp"
+#include "ParallelRestart.hpp"
+#include <ebos/eclmpiserializer.hh>
+
+namespace Opm {
+
+
+ParallelFieldPropsManager::ParallelFieldPropsManager(FieldPropsManager& manager)
+    : m_manager(manager)
+    , m_comm(Dune::MPIHelper::getCollectiveCommunication())
+{
+}
+
+
+std::vector<int> ParallelFieldPropsManager::actnum() const
+{
+    if (m_comm.rank() == 0)
+        return m_manager.actnum();
+
+    return{};
+}
+
+
+void ParallelFieldPropsManager::reset_actnum(const std::vector<int>& actnum)
+{
+    assert(m_comm.rank() == 0);
+    m_manager.reset_actnum(actnum);
+}
+
+
+std::vector<double> ParallelFieldPropsManager::porv(bool global) const
+{
+    std::vector<double> result;
+    if (m_comm.rank() == 0)
+        result = m_manager.porv(global);
+    size_t size = result.size();
+    m_comm.broadcast(&size, 1, 0);
+    result.resize(size);
+    m_comm.broadcast(result.data(), size, 0);
+    return result;
+}
+
+
+const std::vector<int>& ParallelFieldPropsManager::get_int(const std::string& keyword) const
+{
+    auto it = m_intProps.find(keyword);
+    if (it == m_intProps.end())
+        OPM_THROW(std::runtime_error, "No integer property field: " + keyword);
+
+    return it->second;
+}
+
+std::vector<int> ParallelFieldPropsManager::get_global_int(const std::string& keyword) const
+{
+    std::vector<int> result;
+    if (m_comm.rank() == 0)
+        result = m_manager.get_global_int(keyword);
+    size_t size = result.size();
```

Comment:
So this is still called on non-root processes? Would you mind telling us where?

In that case broadcast are becoming quite common.

Reply:
right now, as an intermediate step, all properties which are currently accessed using global indexing in the simulator still do so, and then we do this broadcast. i have follow-up commits where we step by step change to the compressed properties. but step by step.

Reply:
Thanks for the clarifictation. Resolving here.